### DISCUSSION TITLE
Does / could rqlite support extensions - e.g. https://github.com/asg017/sqlite-vec?

### DISCUSSION THREAD

**Discussion by a-h** [rqlite/rqlite#1839] [General]
https://github.com/asg017/sqlite-vec is a plugin that you can load using https://www.sqlite.org/loadext.html

When I run `rqlite -H localhost -p 4001 -u admin:secret` and try to load the plugin with `.load ./vec0`, I get:

```
localhost:4001> .load ./vec0
Error: near ".": syntax error
```

When I run `.help`, I can see that `rqlite` doesn't support the load function.

I can't see anything in the options of `rqlited` that would let me run an initialization script at node first startup etc. where I could install and validate sqlite extensions (all nodes would have to have the same set of plugins installed in order to function as expected).

Is this so niche that no-one else would want it?

**Comment by otoolep** [comment]
rqlite doesn't support extensions at this time, but it's something I would consider for the future. There is some higher priority work I need to complete first.

**Comment by otoolep** [comment]
Looking into this here: https://github.com/rqlite/rqlite/pull/1844

**Comment by otoolep** [comment]
@a-h -- would you be interested in trying out a version of rqlite which allows extensions to be loaded? It would require you to build rqlite from source (which is not difficult). 

https://rqlite.io/docs/install-rqlite/building-from-source/

**Comment by otoolep** [comment]
Extension support is now available in v8.27.0.

See https://rqlite.io/docs/guides/extensions/

**Comment by otoolep** [comment]
https://www.philipotoole.com/rqlite-8-27-loadable-sqlite-extensions-support/ -- shows the extension you asked about being loaded into rqlite.
